src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Processors/GeneralizeProcessor.cs

```diff
@@ -19,7 +19,14 @@ public ProcessResult Process(ElementNode node, ProcessContext context = null, Di
             EnsureArg.IsNotNull(settings);
 
             var result = new ProcessResult();
-            if (!ModelInfo.IsPrimitive(node.InstanceType) || node.Value == null)
+            if (!ModelInfo.IsPrimitive(node.InstanceType))
+            {
+                throw new AnonymizerRuleNotApplicableException(
+                    $"Generalization is not applicable on the node with type {node.InstanceType}. " +
+                    $"Only FHIR primitive nodes (ref: https://www.hl7.org/fhir/datatypes.html#code) are applicable.");
```

Comment:
🟡 Suggestion
The link points to the definition of `code` , but generalize method can apply on more data types. We may use a more general link here.

Reply:
Fixed it.